From 812194e9099934afc2717ae484f94625ea3c0991 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:05:20 -0400
Subject: [ethernet] qlcnic: Do not force adapter to perform LRO without destination IP check

Message-id: <1408626422-3941-37-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88656
O-Subject: [PATCH RHEL 7.1 036/138] qlcnic: Do not force adapter to perform LRO without destination IP check
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shahed Shaikh <shahed.shaikh@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 598c45b309eb401510653fed45fe74efae93be4e

Forcing adapter to perform LRO without destination IP check
degrades the performance.

Signed-off-by: Shahed Shaikh <shahed.shaikh@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c
index 73e72eb..6f7f60c 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c
@@ -785,8 +785,6 @@ void qlcnic_82xx_config_intr_coalesce(struct qlcnic_adapter *adapter)
 
 #define QLCNIC_ENABLE_IPV4_LRO  1
 #define QLCNIC_ENABLE_IPV6_LRO  2
-#define QLCNIC_NO_DEST_IPV4_CHECK (1 << 8)
-#define QLCNIC_NO_DEST_IPV6_CHECK (2 << 8)
 
 int qlcnic_82xx_config_hw_lro(struct qlcnic_adapter *adapter, int enable)
 {
@@ -806,11 +804,10 @@ int qlcnic_82xx_config_hw_lro(struct qlcnic_adapter *adapter, int enable)
 
  word = 0;
  if (enable) {
-  word = QLCNIC_ENABLE_IPV4_LRO | QLCNIC_NO_DEST_IPV4_CHECK;
+  word = QLCNIC_ENABLE_IPV4_LRO;
   if (adapter->ahw->extra_capability[0] &
       QLCNIC_FW_CAP2_HW_LRO_IPV6)
-   word |= QLCNIC_ENABLE_IPV6_LRO |
-    QLCNIC_NO_DEST_IPV6_CHECK;
+   word |= QLCNIC_ENABLE_IPV6_LRO;
  }
 
  req.words[0] = cpu_to_le64(word);
-- 
1.7.1